ivy/functional/backends/paddle/experimental/sorting.py

```diff
@@ -19,6 +19,13 @@ def invert_permutation(
 def lexsort(
     keys: paddle.Tensor, /, *, axis: int = -1, out: Optional[paddle.Tensor] = None
 ) -> paddle.Tensor:
+    if keys.dtype in [
```

Comment:
Why are we upcasting these? :slightly_smiling_face: 
Can't we handle this by adding the `unsupported_dtypes` decorators above this function?

Reply:
According to our discussion as the dtypes suppported by this are `[int32, int64, float32, float64]`. Adding a logic here where we upcast to the closest possible `dtype` would make more sense I guess, what you can do is upcast to `float32` if its a float `dtype` and similarly cast to a closest `int` `dtype` if we get an unsupprted `int` dtype. We will actually be handling a broader set in this way. 
Hope that makes sense :+1: